lib/src/app.dart

```diff
@@ -314,6 +318,31 @@ class _EntryPointState extends ConsumerState<_EntryPointWidget> {
         }
       }
     });
+
+    ref.read(localNotificationServiceProvider).init();
+
+    ref.listenManual(challengesProvider, (prev, current) {
```

Comment:
You should not listen to a provider that should be only linked to a specific view to achieve that. (Cf. other comment of challengesProvider).

`challenges` socket event is a global event, which means it can appear on any socket route of lichess. The SocketPool already makes sure the app is connected to a socket, so you don't need to care about which socket it is connected to. What you want here is to listen to the [global socket event stream](https://github.com/lichess-org/mobile/blob/ba871fe83aeffd16da6878f0fc994420f41c84f9/lib/src/model/common/socket.dart#L52). (You'll need to add 'messages' to the white list).

You could listen to the stream here, but it is probably better to do that in the `challengeService` (where you'd start listening in the `init` method).